Add a Callpower CTA

Pomax opened the pull request:
Will close https://github.com/mozilla/foundation.mozilla.org/issues/7284 when finalized.

Demo URL: https://foundation-s-genericise-ftncey.mofostaging.net/en/campaigns/initial-test-bannered-campaign-with-fixed-title/

edit page: https://foundation-s-genericise-ftncey.herokuapp.com/cms/pages/76/edit/
snippet edit page: https://foundation-s-genericise-ftncey.herokuapp.com/cms/snippets/wagtailpages/callpower/edit/27/

## Testing

there are several "secret" buttons for toggling the various states, because we don't have a test campaign set up atm that allows actual calls to go through, so:

1. Confirm that phone number and zipcode have validation: phone number should only allow a 10 digit US phone number, and zipcode should only allow a five digit US zipcode. You should get errors if that's not the case and you press "make the call".
2. the "show thank you" button should trigger what happens when calling works and the callpower API accepts the make-a-call button click.
3. the "bad fields" button should trigger the "you filled in a phone number we can't call" flow, keeping users on the form but showing them an error text. Once the user edits the phone number or zipcode, this text should disappear again.
4. the "rate limited" button should trigger what happens when the user's placed too many calls per hour already
5. the "show error" button should trigger what happens when things go horrible wrong and we have no idea why.

## Testing against staging/production

To make sure the migrations don't mess with preexisting data, please also test this with the staging database:
1. `inv copy-stage-db`
2. fire up `docker-compose up` and head over to http://localhost:8000/campaigns
3. in a separate tab, open https://foundation.mozilla.org/campaigns
4. open a bunch of campaigns on both, and verify that anything with a petition looks the same between localhost and staging. If they don't, that's a problem =)

## Notes

- the testing buttons must be removed before merging

Pomax commented:
@TheoChevalier added you just to make sure I didn't miss an obvious place where things should be translated but I forgot to add the appropriate code for that.

kristinashu commented:
Yes, that looks good Pomax!

kristinashu commented:
The text looks good. But please also make the form field outline red when there are errors in them so it aligns with the petition form.

- [ ] Update red form field outline when there is an error 

Screen cap from PR
<img width="378" alt="image" src="https://user-images.githubusercontent.com/1068783/137356462-e0e9443a-2c54-4dcd-9f71-bd3478e1a575.png">

Screen cap from petition form on prod
<img width="382" alt="image" src="https://user-images.githubusercontent.com/1068783/137356889-b829a506-4209-4071-a53a-b35e58831a8a.png">

Pomax commented:
@richbrennan reflagging you for review, I updated the migration file to use the pk rather than title/header combination, since pk worked perfeclty fine, but I also made it much easier for you to do local verifcation against staging - I've added some additional instructions to the first comment.

Pomax commented:
@fessehaye reflagged you just to make sure the now final JS makes sense =)
